src/Init/Data/Int/Pow.lean

```diff
@@ -41,4 +42,22 @@ theorem natCast_pow (b n : Nat) : ((b^n : Nat) : Int) = (b : Int) ^ n := by
   | n + 1 =>
     simp only [Nat.pow_succ, Int.pow_succ, natCast_mul, natCast_pow _ n]
 
+theorem natCast_two_pow (n : Nat) : ((2^n : Nat) : Int) = (2: Int) ^ n := by
```

Comment:
```suggestion
theorem natCast_two_pow (n : Nat) : ((2^n : Nat) : Int) = (2 : Int) ^ n := by
```

Reply:
You can just replace `2` with `k : Nat` here, and the proof still works.

Reply:
Indeed, `natCast_pow k n` works as the entire proof!

Reply:
Dropped.